src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4763,7 +4763,7 @@ trait Types
   /** def isNonValueType(tp: Type) = !isValueElseNonValue(tp) */
 
   def isNonRefinementClassType(tpe: Type) = tpe match {
-    case SingleType(_, sym) => sym.isModuleOrModuleClass
+    case SingleType(_, sym) => sym.isModuleClass
```

Comment:
For the record: this reverts the change here that came from #9279, by moving it locally to `typedClassOf`.